components/AddExercises/AddExercises.jsx

```diff
@@ -5,11 +5,54 @@ import ModalScreenWrapper from '../utils/ModalScreenWrapper';
 
 export default ({ route }) => {
   const exercises = useSelector((state) => state.exercises.exercises);
-  const items = exercises.map((exercise) => ({ ...exercise, subtext: exercise.muscleGroups }));
+  const exerciseObjects = exercises.map((exercise) => ({ ...exercise, subtext: exercise.muscleGroups }));
+
+  const parseItemsByName = (items) => {
+    // Sort names alphabetically
+    items.sort((a, b) => a.name.localeCompare(b.name));
+
+    // Group by first letter of each name
+    const bucketData = items.reduce((accumulator, item) => {
+      const bucket = item.name[0].toUpperCase();
+
+      // If this is the first time we've seen this letter, create a bucket
+      if (!accumulator[bucket]) {
+        accumulator[bucket] = [item];
+      } else {
+        accumulator[bucket].push(item);
+      }
+      return accumulator;
+    }, {});
+    return bucketData;
+  };
+
+  const parseItemsByMuscleGroup = (items) => {
+    // Sort by muscle group
+    items.sort((a, b) => a.muscleGroups.localeCompare(b.muscleGroups));
+    const bucketData = items.reduce((accumulator, item) => {
+      const bucket = item.muscleGroups;
+      if (!accumulator[bucket]) {
+        accumulator[bucket] = [item];
+      } else {
+        accumulator[bucket].push(item);
+      }
+      return accumulator;
+    }, {});
+
+    // Sort by name within each muscle group
+    Object.keys(bucketData).forEach((key) => {
+      bucketData[key].sort((a, b) => a.name.localeCompare(b.name));
+    });
+    return bucketData;
+  };
 
   return (
     <ModalScreenWrapper>
-      <ExerciseList items={items} onExercisesAdd={route.params.onExercisesAdd} />
+      <ExerciseList
+        onExercisesAdd={route.params.onExercisesAdd}
```

Comment:
nit: can we add this to props validation or use _.get to make sure we don't error out if route is undefined